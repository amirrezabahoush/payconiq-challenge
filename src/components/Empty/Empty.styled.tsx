import styled from "styled-components";

export const EmptyWrapper = styled.div`
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  padding: 10px;
`;

export const EmptyContent = styled.span`
  display: inline-block;
  background: #f2f5f6;
  color: #5890bd;
  -webkit-border-radius: 4px;
  -moz-border-radius: 4px;
  border-radius: 20px;
  padding: 5px 15px;
`;
