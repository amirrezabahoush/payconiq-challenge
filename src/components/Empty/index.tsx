import React from "react";
import { EmptyContent, EmptyWrapper } from "./Empty.styled";

const Empty = (props: { text: string }) => {
  return (
    <EmptyWrapper>
      <EmptyContent className="empty-div-content">{props.text}</EmptyContent>
    </EmptyWrapper>
  );
};

export default Empty;
