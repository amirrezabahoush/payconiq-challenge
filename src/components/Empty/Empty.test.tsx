import { render, screen } from "@testing-library/react";
import Empty from "./index";

describe("Empty component unit tests", () => {
  it("should render the Empty correctly", () => {
    render(<Empty text="There is no data" />);
    const emptyText = screen.getByText("There is no data");
    expect(emptyText).toBeInTheDocument();
    expect(emptyText).toBeVisible();
  });

  it("should be match with the snapshot", () => {
    render(<Empty text="There is no data" />);
    const emptyText = screen.getByText("There is no data");
    expect(emptyText).toMatchSnapshot();
  });
});
