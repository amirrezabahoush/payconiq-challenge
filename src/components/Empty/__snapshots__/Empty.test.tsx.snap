// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty component unit tests should be match with the snapshot 1`] = `
<span
  class="sc-gsnTZi jJapeB empty-div-content"
>
  There is no data
</span>
`;
